crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -55,6 +59,11 @@ def checks
             passed: clusters_healthy?,
             errors: clusters_health_check_errors
           } if Api::Crowbar.addons.include?("ha")
+
+          return ret unless upgrade_status.current_step == "upgrade_prechecks"
+
+          errors = ret.select { |_k, v| v[:required] && v[:errors].any? }.map { |_k, v| v[:errors] }
+          upgrade_status.end_step(errors.any?, errors)
```

Comment:
We need to call `upgrade_status.end_step(errors.empty?, errors)`, as empty == success